src/main.rs

```diff
@@ -51,6 +52,16 @@ fn main() -> Result<(), Error> {
 
     let rtsp = &RtspServer::new();
 
+    let tls_client_auth = match &config.tls_client_auth as &str {
+        "request" => TlsAuthenticationMode::Requested,
+        "require" => TlsAuthenticationMode::Required,
+        "none"|_ => TlsAuthenticationMode::None,
+    };
```

Comment:
I'm not a fan of defaulting `_` to None. Say there was a typo `reqiure`. The user doesn't intend for it to be none.

Now, you do have a regex protecting this. So make the default match branch go

```rust
_ => unreachable!()
```

which is a macro that will panic if it's ever hit but documents that you think it's unreachable.

Reply:
Overall I think a cleaner way to do one-of config values like this would be to have an enum and let the config parser figure out which variant it is, but I don't have much experience with that so this approach is fine.

Reply:
Yes I was thinking it didn't matter because of the regex but this is better